packages/builders/src/components/ActionRow.ts

```diff
@@ -1,68 +1,60 @@
+import type { AnyActionRowComponentBuilder } from './Components.js';
 import { createComponentBuilder } from './Components.js';
-import type { ButtonBuilder } from './button/Button.js';
-import type { ChannelSelectMenuBuilder } from './selectMenu/ChannelSelectMenu.js';
-import type { MentionableSelectMenuBuilder } from './selectMenu/MentionableSelectMenu.js';
-import type { RoleSelectMenuBuilder } from './selectMenu/RoleSelectMenu.js';
-import type { StringSelectMenuBuilder } from './selectMenu/StringSelectMenu.js';
-import type { UserSelectMenuBuilder } from './selectMenu/UserSelectMenu.js';
-import type { TextInputBuilder } from './textInput/TextInput.js';
-
-/**
- * The builders that may be used for messages.
- */
-export type MessageComponentBuilder =
-	| ActionRowBuilder<MessageActionRowComponentBuilder>
-	| MessageActionRowComponentBuilder;
-
-/**
- * The builders that may be used for modals.
- */
-export type ModalComponentBuilder = ActionRowBuilder<ModalActionRowComponentBuilder> | ModalActionRowComponentBuilder;
-
-/**
- * The builders that may be used within an action row for messages.
- */
-export type MessageActionRowComponentBuilder =
-	| ButtonBuilder
-	| ChannelSelectMenuBuilder
-	| MentionableSelectMenuBuilder
-	| RoleSelectMenuBuilder
-	| StringSelectMenuBuilder
-	| UserSelectMenuBuilder;
-
-/**
- * The builders that may be used within an action row for modals.
- */
-export type ModalActionRowComponentBuilder = TextInputBuilder;
+import {
+	DangerButtonBuilder,
+	PrimaryButtonBuilder,
+	SecondaryButtonBuilder,
+	SuccessButtonBuilder,
+} from './button/CustomIdButton.js';
+import { LinkButtonBuilder } from './button/LinkButton.js';
+import { PremiumButtonBuilder } from './button/PremiumButton.js';
+import { ChannelSelectMenuBuilder } from './selectMenu/ChannelSelectMenu.js';
+import { MentionableSelectMenuBuilder } from './selectMenu/MentionableSelectMenu.js';
+import { RoleSelectMenuBuilder } from './selectMenu/RoleSelectMenu.js';
+import { StringSelectMenuBuilder } from './selectMenu/StringSelectMenu.js';
+import { UserSelectMenuBuilder } from './selectMenu/UserSelectMenu.js';
+import { TextInputBuilder } from './textInput/TextInput.js';
 
-/**
- * Any builder.
- */
-export type AnyComponentBuilder = MessageActionRowComponentBuilder | ModalActionRowComponentBuilder;
+export interface ActionRowBuilderData
+	extends Partial<Omit<APIActionRowComponent<APIActionRowComponentTypes>, 'components'>> {
+	components: AnyActionRowComponentBuilder[];
+}
 
 /**
  * A builder that creates API-compatible JSON data for action rows.
  *
  * @typeParam ComponentType - The types of components this action row holds
  */
-export class ActionRowBuilder<ComponentType extends AnyComponentBuilder> extends ComponentBuilder<
-	APIActionRowComponent<APIMessageActionRowComponent | APIModalActionRowComponent>
-> {
+export class ActionRowBuilder extends ComponentBuilder<APIActionRowComponent<APIActionRowComponentTypes>> {
```

Comment:
There should be a way to add buttons of any style. Say we are mapping an array to builders or already have a preset array of builders